File: Python/maxProfit.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0
        price1 = 0
        price2 = 1

        while price2 < len(prices):
            if profit < (prices[price2]-prices[price1]):
                profit = prices[price2]-prices[price1]
            
            if prices[price1] > prices[price2]: 
                price1 = price2
            price2 += 1
        return profit

File: Java/maxProfit.java

class Solution {
    public int maxProfit(int[] prices) {
        int max_profit = 0, min_day = prices[0];

        for (int x : prices) {
            if (x < min_day) { min_day = x; }
            
            if (max_profit < (x - min_day)) {
                max_profit = x - min_day; 
            }
        }
        return max_profit;
    }
}

class Main {
    public static void main(String[] args) {
        int[] prices = {7,6,4,3,1};
        Solution solution = new Solution();
        System.out.print(solution.maxProfit(prices));
    }
}